tests/README.md

```diff
@@ -1,25 +1,239 @@
+$ pip3 install -r requirements/requirements/python-requirements.txt
+```
+
+
+-------------------------------------------------------------------------------
+
+
+## Running the Tests Locally
+
+> The tests can be run locally without any further involvement as long as all
+> the dependencies have been installed and are at the correct version. However,
+> managing dependencies, especially with Python can be a hassle. Therefore it is
+> recommended to add a virtual Python environment to isolate the dependencies
+> needed for running the integration tests.
+
+### Isolating Python Dependencies
+
+In order to avoid dependency mismanagement due to Python packages differing from
+one test environment to the other, it is recommended to use a python virtual
+environment. The "de-facto" standard is
+[virtualenv](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/).
+
+#### Install
+
+```bash
+$ pip install virtualenv
+```
+
+#### Initialize the Virtual Environment
+
+```bash
+$ cd integration-dir/tests
+$ virtualenv <name-of-virtualenv-folder>
+```
+
+##### (Optional) -- Select Python Version
+```bash
+$ virtualenv -p /usr/bin/python2.7 <name-of-virtualenv-folder>
+```
+
+#### Activate the Virtual Environment
+
+```bash
+$ source <name-of-virtualenv-folder>/bin/activate
+```
+
+This now means that you have a clean python environment, and no packages you
+have previously installed outside of this virtual environment will be
+discoverable by Python.
+
+#### Install the Python Dependencies from the Requirements File
+
+```bash
+$ pip install -r requirements/integration/python-requirements.txt
+```
+
+This will install the requirements needed to run the tests.
+
+### Running the Tests
+
+Next, run the tests with the `run.sh` script.
+
+```bash
+$ ./run.sh
+```
+
+**NOTE**: This is dependent upon having a functioning Docker environment, and being
+logged in to `registry.mender.io`.
+
+## Modifying the Docker Images Employed
+
+In order to run the integration tests with the local changes made to some Mender
+service, it is necessary to build the container, and tag it with the matching
+tag employed in the integration repository. This can be found in the
+`docker-compose.yml` file in the root directory, under the `image:` key.
+
+#### Example -- Running with a Custom Backend Service
+
+For the backend services there exists a `Dockerfile` in the root repository, and
+as such a new image can be built and tagged by
+
+```bash
+$ cd /path/to/<service>/
+$ docker build . -t mendersoftware/<service>:master
+```
+
+#### Example -- Running with a Custom Client
+
+For building a custom client the approach is a little different, due to the fact
+that the client comes bundles with a Yocto image. Therefore, in order to build
+and run a custom client with the integration test setup, first build a Yocto
+image, containing the custom client. Then, build a Docker image containing this
+client by
+
+```bash
+$ cd /path/to/meta-mender
```

Comment:
Just to make it crystal clear, I would add the `source oe-init-build-env` as an explicit step here. It doesn't hurt to do it more than once, and might save some confusion (people like to skim read!).

Reply:
Following Kristian comment, I would merge the two bunch of steps into a single one:
```
cd /path/to/yocto/dir
source oe-init-build-env
bitbake ... <- The full image, I don't remember now the exact name
cd ../meta-mender
cd meta-mender-qemu/docker
./build-docker qemux86-64 -t mendersoftware/mender-client-qemu:master
```